src/main/kotlin/com/pubnub/api/PubNub.kt

```diff
@@ -949,6 +953,40 @@ class PubNub(val configuration: PNConfiguration) {
         )
     }
 
+    /**
+     * This function generates a grant token for PubNub Access Manager (PAM).
+     *
+     * Permissions can be applied to any of the two type of resources:
+     * - spaces
+     * - userIds
+     *
+     * Each type of resource have different set of permissions. To know what's possible for each of them
+     * check SpaceIdGrant and UserIdGrant.
+     *
+     * @param ttl Time in minutes for which granted permissions are valid.
+     * @param meta Additional metadata
+     * @param authorizedUserId Single userId which is authorized to use the token to make API requests to PubNub
+     * @param spaces List of all channel grants
+     * @param userIds List of all uuid grants
+     */
+    fun grantToken(
+        ttl: Int,
+        meta: Any? = null,
+        authorizedUserId: UserId? = null,
+        spaces: List<SpaceIdGrant> = emptyList(),
+        userIds: List<UserIdGrant> = emptyList()
```

Comment:
`spaces` and `usersIds`? Would be probably better to have one way of naming it

Reply:
yes, sure.  I forgot to change spaces to spaceIds.

Reply:
maybe space/user permissions? I mean it's not only id, right?

Reply:
I like UserPermissions and SpacePermissions. I will change it.